placeholder-transaction-trait/src/message_trait.rs

```diff
@@ -0,0 +1,217 @@
+                if let Some(num_verifies) = instruction.data.first() {
+                    transaction_signature_details.num_secp256k1_instruction_signatures =
+                        transaction_signature_details
+                            .num_secp256k1_instruction_signatures
+                            .saturating_add(u64::from(*num_verifies));
+                }
+            } else if solana_sdk::ed25519_program::check_id(program_id) {
+                if let Some(num_verifies) = instruction.data.first() {
+                    transaction_signature_details.num_ed25519_instruction_signatures =
+                        transaction_signature_details
+                            .num_ed25519_instruction_signatures
+                            .saturating_add(u64::from(*num_verifies));
+                }
+            }
+        }
+
+        transaction_signature_details
+    }
+
+    /// Return the durable nonce for the message if it exists
+    fn get_durable_nonce(&self) -> Option<&Pubkey> {
+        self.instructions_iter()
+            .nth(NONCED_TX_MARKER_IX_INDEX as usize)
+            .filter(
+                |ix| match self.account_keys().get(ix.program_id_index as usize) {
+                    Some(program_id) => solana_sdk::system_program::check_id(program_id),
+                    _ => false,
+                },
+            )
+            .filter(|ix| {
+                matches!(
+                    solana_program::program_utils::limited_deserialize(
+                        ix.data, 4 /* serialized size of AdvanceNonceAccount */
+                    ),
+                    Ok(solana_sdk::system_instruction::SystemInstruction::AdvanceNonceAccount)
+                )
+            })
+            .and_then(|ix| {
+                ix.accounts.first().and_then(|idx| {
+                    let idx = *idx as usize;
+                    if !self.is_writable(idx) {
+                        None
+                    } else {
+                        self.account_keys().get(idx)
+                    }
+                })
+            })
+    }
+
+    /// Return the signers for the instruction at the given index.
+    fn get_ix_signers(&self, index: usize) -> impl Iterator<Item = &Pubkey> {
+        self.instructions_iter()
+            .nth(index)
+            .into_iter()
+            .flat_map(|ix| {
+                ix.accounts
+                    .iter()
+                    .copied()
+                    .map(usize::from)
+                    .filter(|index| self.is_signer(*index))
+                    .filter_map(|signer_index| self.account_keys().get(signer_index))
+            })
+    }
+
+    /// Checks for duplicate accounts in the message
+    fn has_duplicates(&self) -> bool;
+
+    /// Decompile message instructions without cloning account keys
+    /// TODO: Remove this - there's an allocation!
+    fn decompile_instructions(&self) -> Vec<BorrowedInstruction> {
```

Comment:
note to self: can drop this if we replace the single-use inside of `account_loader::construct_instructions_account`; This seems easy to do if we just iterate over the instructions if also given number of instructions.

Reply:
removed this; can easily have a free-function that replaces the needed functionality, want to keep the interface here to a minimum.